Controller/RegistrationFOSUser1Controller.php

```diff
@@ -56,10 +56,10 @@ public function registerAction()
                 $url = $this->generateUrl('fos_user_registration_check_email');
             } else {
                 $authUser = true;
-                $route = $this->get('session')->get('sonata_basket_delivery_redirect');
+                $route = $this->get('session')->get('sonata_user_redirect_url');
 
                 if (null !== $route) {
-                    $this->get('session')->remove('sonata_basket_delivery_redirect');
+                    $this->get('session')->remove('sonata_user_redirect_url');
                     $url = $this->generateUrl($route);
                 } else {
                     $url = $this->get('session')->get('sonata_user_redirect_url', $this->generateUrl('sonata_user_profile_show'));
```

Comment:
Let's translate in pseudo-code : 

```
get the user redirect url in session

if it is not null
	remove it from the session
	generate an url from it (so it is in fact a route, not an url)
else
	try to get it again anyway, and default to the profile show route
```

Sound pretty stupid, right?  :stuck_out_tongue: 

Let 's just do 
```php
<?php
$url = $this->get('session')->get(
    'sonata_user_redirect_url',
     $this->generateUrl('sonata_user_profile_show')
);
// and maybe this too
$this->get('session')->remove('sonata_user_redirect_url');
```